File: C# 2/Arrays/MaximalIncreasingSequence/MaximalIncreasingSequence.cs
// Problem 5. Maximal increasing sequence
// Write a program that finds the maximal increasing sequence in an array.


using System;

class MaximalIncreasingSequence
{
    static void Main()
    {
        Console.WriteLine("enter the length of the array");
        int length = int.Parse(Console.ReadLine());
        int[] arr = new int[length];
        for (int i = 0; i < length; i++)
        {
            arr[i] = int.Parse(Console.ReadLine());
        }

        int currentNumber = arr[0];
        int sequenceLength=0;
        int bestLength = 0;
        int bestNumber = 0;


        for(int i=1;i<length;i++)
        {   
            
            if(currentNumber==arr[i]-1)
            {
                currentNumber = arr[i];
                sequenceLength++;
            }
            else
            {
                currentNumber = arr[i];
                sequenceLength = 0;
            }

            if(sequenceLength>bestLength)
            {
                bestLength = sequenceLength;
                bestNumber = currentNumber;
            }
        }

        for(int i=0;i<=bestLength;i++)
        {
            Console.Write(bestNumber-bestLength+i+" ");
        }
    }
}

File: JS/Arrays/MaximalIncreasingSequence.js
// Write a script that finds the maximal increasing sequence in an array.

var arr = [2, 2, 3, 4, 2, 2, 4,5],
    i = 0,
    j = 0,
    count = 0,
    maxCount = 0,
    digit;

for(i; i < arr.length-1;i+=1) {
    if(arr[i]===arr[i+1]-1) {
        count+=1;
        if(count>maxCount) {
            maxCount = count;
            digit=arr[i];
        }
    } else {
        count =0;
    }
}

for(j; j < maxCount+1;j+=1) {
     console.log(digit-maxCount+j+1);
}